pkg/kubelet/eviction/interfaces.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eviction
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/resource"
+)
+
+// Signal defines a signal that can trigger eviction of pods on a node.
+type Signal string
+
+const (
+	// SignalMemoryAvailable is memory available (i.e. capacity - workingSet), in bytes.
+	SignalMemoryAvailable Signal = "memory.available"
+)
+
+// ThresholdOperator is the operator used to express a Threshold.
+type ThresholdOperator string
+
+const (
+	// OpLessThan is the operator that expresses a less than operator.
+	OpLessThan ThresholdOperator = "LessThan"
+	// OpGreaterThan is the operator that expresses a greater than operator.
+	OpGreaterThan ThresholdOperator = "GreaterThan"
+)
+
+// Threshold defines a metric for when eviction should occur.
+type Threshold struct {
```

Comment:
So I wanted to avoid passing 5 additional flags down to `NewMainKubelet`.

This seemed consistent with how we handle reservations.

see: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L217
which is: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/types/types.go#L72